zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationCatchEventBehaviour.java

```diff
@@ -174,6 +174,20 @@ private void migrateCompensationEvents(
           recordCopy.setProcessDefinitionKey(targetProcessDefinition.getKey());
           recordCopy.setCompensableActivityId(targetActivityId);
 
+          // set the compensation handler id if subscription belongs to an activity with a boundary
+          targetProcessDefinition
```

Comment:
🔧 There is a method in the CompensationSubscriptionBehaviour that return the handlerId given an `ExecutableActivity`. You can reuse it